log/log.go

```diff
@@ -64,25 +64,33 @@ var bufStdErr bytes.Buffer
 // that can be used for checking what the logger would've written. This is
 // mostly used for tests.
 func OutputToBuf() {
+	debugMut.Lock()
+	defer debugMut.Unlock()
 	stdOut = &bufStdOut
 	stdErr = &bufStdErr
 }
 
 // OutputToOs redirects the output of the log.*-outputs again to the os.
 func OutputToOs() {
+	debugMut.Lock()
+	defer debugMut.Unlock()
 	stdOut = os.Stdout
 	stdErr = os.Stderr
 }
 
 // GetStdOut returns all log.*-outputs to StdOut since the last call.
 func GetStdOut() string {
+	debugMut.Lock()
+	defer debugMut.Unlock()
 	ret := bufStdOut.String()
 	bufStdOut.Reset()
 	return ret
 }
 
 // GetStdErr returns all log.*-outputs to StdErr since the last call.
 func GetStdErr() string {
```

Comment:
Ok ok I won't insist more, you decide if you want to not have this method it's fine. There's one counter argument in favor of your approach is that `ContainsStdOut` do not say that it flushes the buffer also, and that might be an unwanted behavior.
So it's fine if you retract the last commit or not for me then, just tell me and I merge.